### PR TITLE
Relax `package/version` check for multi `outputs`

### LINKED ISSUE #5097: Recipes defining `outputs` without `package/version` fail in 3.28.0
### What happened?

https://github.com/conda/conda-build/pull/5055 corrected how the metadata's version was validated and resulted in multi outputs failing to build unless they also define the top level `package/version`.

Xref https://github.com/conda-forge/admin-requests/pull/890

### Conda Details

<details>
<summary><code>conda info</code></summary>

```
     active environment : None
            shell level : 0
       user config file : /Users/kodegard/.condarc
 populated config files : /Users/kodegard/.condarc
          conda version : 23.11.0
    conda-build version : 3.28.0
         python version : 3.10.13.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=m1
                          __conda=23.11.0=0
                          __osx=13.6=0
                          __unix=0=0
       base environment : /Users/kodegard/.conda/arm64/23.10.0/3.11  (writable)
      conda av data dir : /Users/kodegard/.conda/arm64/23.10.0/3.11/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-arm64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-arm64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/kodegard/.conda/arm64/23.10.0/3.11/pkgs
                          /Users/kodegard/.conda/pkgs
       envs directories : /Users/kodegard/.conda/arm64/23.10.0/3.11/envs
                          /Users/kodegard/.conda/envs
               platform : osx-arm64
             user-agent : conda/23.11.0 requests/2.31.0 CPython/3.10.13 Darwin/22.6.0 OSX/13.6 solver/libmamba conda-libmamba-solver/23.11.1 libmambapy/1.5.3
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```

</details>

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Correcting conda-build issues identified for multi-outputs where the version is undefined (https://github.com/conda-forge/admin-requests/pull/890).

Leaving in the version checks in simple recipes but relaxes the requirements for recipes with multi-outputs.

Xref https://github.com/conda/conda-build/pull/5055
Resolves #5097 

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: conda_build/metadata.py [hunk 1]
```diff
@@ -1442,7 +1442,7 @@ def name(self) -> str:
 
     def version(self) -> str:
         version = self.get_value("package/version", "")
-        if not version and self.final:
+        if not version and not self.get_section("outputs") and self.final:
             sys.exit("Error: package/version missing in: %r" % self.meta_path)
```
Comment (dholth) [inline conda_build/metadata.py:1446]: I guess this was there before, and not "raise some CondaBuildError" 🦺

**Review by dholth — APPROVED** [review]
Thanks

File: tests/test_api_build.py [hunk 2]
```diff
@@ -412,6 +412,9 @@ def dummy_executable(folder, exename):
     return exename
 
 
+@pytest.mark.skip(
+    reason="GitHub discontinued SVN, see https://github.com/conda/conda-build/issues/5098"
+)
```
Comment (kenodegard) [inline tests/test_api_build.py:417]: GitHub has started the processes of turning off SVN support causing this tests to fail. Refactoring revealed unexpected surprises, so I've chosen to step back for this hotfix and temporarily bypass the failing test.

See https://github.com/conda/conda-build/issues/5098 for more.

File: conda_build/metadata.py (hunk 1 above)
Comment (kenodegard) [reply conda_build/metadata.py:1446]: Correct

**Review by jezdez — APPROVED** [review]

**Pull request merged**